src/com/goide/inspections/GoStringCannotBeNilInspection.java

```diff
@@ -0,0 +1,95 @@
+package com.goide.inspections;
+
+import com.goide.GoConstants;
+import com.goide.GoTypes;
+import com.goide.psi.*;
+import com.goide.psi.impl.GoElementFactory;
+import com.goide.psi.impl.GoTypeUtil;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.LocalQuickFix;
+import com.intellij.codeInspection.ProblemDescriptor;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiElement;
+import org.jetbrains.annotations.Nls;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class GoStringCannotBeNilInspection extends GoInspectionBase{
+
+  private static final String DEFAULT_STRING = "\"\"";
+  private static final String PROBLEM_DESC = "String  cannot be nil";
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor(){
+
+
+      @Override
+      public void visitVarSpec(@NotNull GoVarSpec o) {
+        super.visitVarSpec(o);
+        for(int i = 0; i < o.getExpressionList().size(); ++i){
+          check(o.getVarDefinitionList().get(i).getGoType(null), o.getVarDefinitionList().get(i).getValue());
+        }
+      }
+
+
+      @Override
+      public void visitAssignOp(@NotNull GoAssignOp o) {
+        super.visitAssignOp(o);
+        PsiElement[] right = o.getParent().getChildren();
```

Comment:
looks cryptic and non-reliable, you cannot say what children of parent are. if you want to visit some particular elements, you should process them, not their children.

Also getParent might be `null`, so yet another NPE here.